### PR TITLE
Improve tracing of `ChainUpdate`s in ChainSync server

### PR DESCRIPTION
**Pull request by amesgen**
Closes [CAD-4251](https://input-output.atlassian.net/browse/CAD-4251)

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs [hunk 1]
```diff
@@ -148,12 +151,20 @@ chainSyncServerForFollower tracer chainDB flr =
...
-data TraceChainSyncServerEvent blk
-  = TraceChainSyncServerRead        (Tip blk) (ChainUpdate blk (Point blk))
-  | TraceChainSyncServerReadBlocked (Tip blk) (ChainUpdate blk (Point blk))
-  | TraceChainSyncRollForward       (Point blk)
-  | TraceChainSyncRollBackward      (Point blk)
+data TraceChainSyncServerEvent blk =
+    -- | Read a chain update instruction non-blockingly.
+    TraceChainSyncServerRead        (TraceChainSyncServerInfo blk)
+    -- | Read a chain update instruction blockingly.
+  | TraceChainSyncServerReadBlocked (TraceChainSyncServerInfo blk)
```
Comment (nfrisby) [inline ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Server.hs:158]: Do we want to make `TraceChainSyncServerReadBlocked` just a boolean-isomorph field in `TraceChainSyncServerRead` while we're at it?

**Review by nfrisby — APPROVED** [review]
Approved. `ouroboros-consensus/docs/interface-CHANGELOG.md` does not yet contain any tracer event changes, but I'm still not certain whether it should or not. For this PR, I figure we don't need to update it, but maybe it'd be neighborly for us to open a matching `cardano-node` PR?

**Comment by amesgen** [comment]
> For this PR, I figure we don't need to update it, but maybe it'd be neighborly for us to open a matching `cardano-node` PR?

Agreed, I will create a consensus bump PR with the respective tracing changes, tracked in [CAD-4312](https://input-output.atlassian.net/browse/CAD-4312).

**Comment by amesgen** [comment]
I will rebase this when #3752 has been merged to use the new `Enclose` mini-abstraction.

**Comment by amesgen** [comment]
bors merge

**Pull request merged**